### PR TITLE
[DOCS] Adds retention_policy to PUT Transform API docs

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds the `retention_policy` property and its sub-properties to the PUT Transform API and update transform API docs. It also expands the example API call with the property in the PUT Transform API docs and the example response in the GET Transform stats API docs.

### Preview

* [PUT Transform API docs](https://elasticsearch_68656.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/put-transform.html)
* [GET Transform stats API docs](https://elasticsearch_68656.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/get-transform-stats.html)
* [UPDATE transform API docs](https://elasticsearch_68656.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/update-transform.html)

### Note

This PR depends on https://github.com/elastic/elasticsearch/pull/67832
Update the branch after that PR is merged to pass the CI tests.

### CODE REVIEW

**Comment by szabosteve** [comment]
@elasticmachine test this please

File: docs/reference/rest-api/common-parms.asciidoc [hunk 1]
```diff
@@ -981,6 +981,24 @@ The `latest` method transforms the data by finding the latest document for each
...
+Defines a retention policy for the {transform}. Data that meets the defined 
+criteria is deleted from the destination index.
+end::transform-retention[]
+
+tag::transform-retention-time[]
+Specifies that the {transform} uses a time field to set the retention policy.
+end::transform-retention-time[]
+
+tag::transform-retention-time-field[]
+The date filed that is used to calculate the age of the document.
```
Comment (hendrikmuhs) [inline docs/reference/rest-api/common-parms.asciidoc:994]: filed -> field

**Review by hendrikmuhs — APPROVED** [review]
1 small typo to be fixed, otherwise LGTM

**Comment by szabosteve** [comment]
@elasticmachine update branch

**Review by lcawl — APPROVED** [review]
LGTM

File: docs/reference/transform/apis/put-transform.asciidoc [hunk 2]
```diff
@@ -151,6 +152,32 @@ include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=pivot-group-by]
...
++
+.Properties of `time`
+[%collapsible%open]
+=====
+`field`:::
+(Required, string)
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=transform-retention-time-field]
+
+`max_age`:::
+(Requided, <<time-units, time units>>)
```
Comment (lcawl) [inline docs/reference/transform/apis/put-transform.asciidoc:175]:
```suggestion
(Required, <<time-units, time units>>)
```

File: docs/reference/transform/apis/update-transform.asciidoc [hunk 3]
```diff
@@ -102,6 +102,32 @@ include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=dest-pipeline]
...
++
+.Properties of `time`
+[%collapsible%open]
+=====
+`field`:::
+(Required, string)
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=transform-retention-time-field]
+
+`max_age`:::
+(Requided, <<time-units, time units>>)
```
Comment (lcawl) [inline docs/reference/transform/apis/update-transform.asciidoc:125]:
```suggestion
(Required, <<time-units, time units>>)
```

Comment (lcawl) [inline docs/reference/transform/apis/update-transform.asciidoc]:
```suggestion
(Required, <<time-units, time units>>)
```

**Pull request merged**